expression/integration_test.go

```diff
@@ -1296,6 +1296,25 @@ func (s *testIntegrationSuite) TestTimeBuiltin(c *C) {
 		"c), subtime(c,a), subtime(c,b) from t;")
 	result.Check(testkit.Rows("<nil> <nil> <nil> 2017-01-01 11:29:30 2017-01-01 11:29:30 <nil> <nil>"))
 
+	// ADDTIME & SUBTIME issue #5966
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a datetime, b timestamp, c time, d date, e bit(1))")
+	tk.MustExec(`insert into t values("2017-01-01 12:30:31", "2017-01-01 12:30:31", "01:01:01", "2017-01-01", 0b1)`)
+
+	result = tk.MustQuery("select addtime(a, e), addtime(b, e), addtime(c, e), addtime(d, e) from t")
+	result.Check(testkit.Rows("<nil> <nil> <nil> <nil>"))
+	result = tk.MustQuery("select addtime('2017-01-01 01:01:01', 0b1), addtime('2017-01-01', b'1'), addtime('01:01:01', 0b1011)")
```

Comment:
@zz-jason Do you test it on MySQL 5.7? I'm pretty sure behavior of MySQL 8.0 has been changed. I think NULL result is more meaningful. Do we still need to conform with mysql 5.7?

Reply:
Yes. For now we only need to be compatible with MySQL 5.7. 😂

Reply:
I just wonder if it deserves to make great effort to be compatible with MySQL 5.7 for the addtime/subtime behavior. I find MySQL 5.7 makes me confusing. For example, addtime('2017-01-01 01:01:01', 0b1) returns '2017-01-01 01:01:01', but addtime('2017-01-01 01:01:01', 0b1011) returns NULL.
Currently TiDB behavior is compatible with MySQL 8.0 except that it can not recognize 0b1. So do you prefer to be compatible with 5.7 now and later rollback?

Reply:
@shenli What's your opinion?